src/lib/Permission/PermissionChecker.php

```diff
@@ -0,0 +1,214 @@
+                    $restrictions[] = $limitation->limitationValues;
+                    $policyHasLimitation = true;
+                }
+            }
+            if (false === $policyHasLimitation) {
+                $oneOfPoliciesHasNoLimitation = true;
+            }
+        }
+
+        if ($oneOfPoliciesHasNoLimitation) {
+            return [];
+        }
+
+        return empty($restrictions) ? $restrictions : array_unique(array_merge(...$restrictions));
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Location $location
+     * @param array|bool $hasAccess
+     *
+     * @return bool
+     */
+    public function canCreateInLocation(Location $location, $hasAccess): bool
+    {
+        if (is_bool($hasAccess)) {
+            return $hasAccess;
+        }
+        $restrictedLocations = $this->getRestrictions($hasAccess, LocationLimitation::class);
+        $canCreateInLocation = empty($restrictedLocations)
+            ? true
+            : in_array($location->id, array_map('intval', $restrictedLocations), true);
+
+        $restrictedParentContentTypes = $this->getRestrictions($hasAccess, ParentContentTypeLimitation::class);
+        $canCreateInParentContentType = empty($restrictedParentContentTypes)
+            ? true
+            : in_array($location->contentInfo->contentTypeId, array_map('intval', $restrictedParentContentTypes), true);
+
+        $restrictedParentDepths = $this->getRestrictions($hasAccess, ParentDepthLimitation::class);
+        $canCreateInParentDepth = empty($restrictedParentDepths)
+            ? true
+            : in_array($location->depth, array_map('intval', $restrictedParentDepths), true);
+
+        $restrictedParentOwner = $this->getRestrictions($hasAccess, ParentOwnerLimitation::class);
+        $canCreateInParentOwner = empty($restrictedParentOwner)
+            ? true
+            : $location->contentInfo->ownerId === $this->permissionResolver->getCurrentUserReference()->getUserId();
+
+        $restrictedSections = $this->getRestrictions($hasAccess, SectionLimitation::class);
+        $canCreateInSection = empty($restrictedSections)
+            ? true
+            : in_array($location->contentInfo->sectionId, array_map('intval', $restrictedSections), true);
+
+        $restrictedParentUserGroups = $this->getRestrictions($hasAccess, ParentUserGroupLimitation::class);
+        $canCreateInParentUserGroup = empty($restrictedParentUserGroups)
+            ? true
+            : $this->hasSameParentUserGroup($location);
+
+        $restrictedSubtrees = $this->getRestrictions($hasAccess, SubtreeLimitation::class);
+        $canCreateInSubtree = empty($restrictedSubtrees)
+            ? true
+            : !empty(array_filter($restrictedSubtrees, function ($restrictedSubtree) use ($location) {
+                return strpos($location->pathString, $restrictedSubtree) === 0;
+            }));
+
+        return $canCreateInParentContentType
```

Comment:
There is no need to check further criterions if `$canCreateInParentContentType` fails, etc. so we should return false immediately without further checks.